pulp_rpm/tests/functional/api/test_publish.py

```diff
@@ -223,6 +223,80 @@ def test_publish_references_update(assert_created_publication):
     assert_created_publication(RPM_REFERENCES_UPDATEINFO_URL)
 
 
+def get_metadata_content_helper(base_url, repomd_elem, meta_type):
+    """Return the text contents of metadata file.
+
+    Provided a url, a repomd root element, and a metadata type, locate the metadata
+    file's location href, download it from the provided url, un-gzip it, parse it, and
+    return the root element node.
+
+    Don't use this with large repos because it will blow up.
+    """
+    # <ns0:repomd xmlns:ns0="http://linux.duke.edu/metadata/repo">
+    #     <ns0:data type="primary">
+    #         <ns0:checksum type="sha256">[…]</ns0:checksum>
+    #         <ns0:location href="repodata/[…]-primary.xml.gz" />
+    #         …
+    #     </ns0:data>
+    #     …
+    xpath = "{{{}}}data".format(RPM_NAMESPACES["metadata/repo"])
+    data_elems = [elem for elem in repomd_elem.findall(xpath) if elem.get("type") == meta_type]
+    if not data_elems:
+        return None
+
+    xpath = "{{{}}}location".format(RPM_NAMESPACES["metadata/repo"])
+    location_href = data_elems[0].find(xpath).get("href")
+
+    return download_and_decompress_file(os.path.join(base_url, location_href))
+
+
+@pytest.mark.parametrize("layout", ["flat", "nested_alphabetically"])
+def test_repo_layout(
+    layout,
+    init_and_sync,
+    rpm_publication_api,
+    gen_object_with_cleanup,
+    rpm_distribution_api,
+    rpm_distribution_factory,
+    monitor_task,
+    delete_orphans_pre,
+    tmpdir,
+    wget_recursive_download_on_host,
+):
+    """Test that using the "layout" option for publication produces the correct package layouts"""
+
+    # create repo and remote
+    repo, _ = init_and_sync(url=RPM_UNSIGNED_FIXTURE_URL, policy="on_demand")
+
+    # publish
+    publish_data = RpmRpmPublication(repository=repo.pulp_href, layout=layout)
+    publish_response = rpm_publication_api.create(publish_data)
+    created_resources = monitor_task(publish_response.task).created_resources
+    publication_href = created_resources[0]
+
+    # distribute
+    body = rpm_distribution_factory(publication=publication_href)
+    distribution = gen_object_with_cleanup(rpm_distribution_api, body)
```

Comment:
`rpm_distribution_factory` returns a client RpmDistributionResponse (or something like that, you can find all those client request/response classes at `pulp-openapi-generator/pulp_rpm-client/` dir.
And all those factory fixtures already creates the resource with the cleanup hook.

So you can call the factory without worryng with the api call/cleanup:
```
dsitribution = rpm_distribution_factory(publication=publication_href)
```